### DISCUSSION TITLE
RenderThread in CARLA Simulator is using 100% of 1 CPU core

### DISCUSSION THREAD

**Discussion by ebinroyabraham** [carla-simulator/carla#4284] [Q&A] [answered]
Hallo,
I am currently using CARLA Simulator in Microsoft Azure VM using Docker container. Please see below the details,
- Azure VM Size: Standard NC4as_T4_v3
- Ubuntu 18.04
- Docker image: carlasim/carla:0.9.10

I executed the CARLA Simulator and connected manual_control.py to the server. The server had around 20-25 FPS and Client 60-62 FPS.
- Is this the expected behaviour from CARLA?

I found out that 1 of the 4 CPU cores was always at 100%. And upon inspection this might be because there was only 1 RenderThread and this is making the usage of 1 core to be at 100%.
- Is it possible to have multiple RenderThread so that it will be distributed to different CPU cores?

Many thanks in advance.

**Comment by bernatx** [comment, marked as answer]
Hi,

By design Unreal uses only one thread for the rendering, so from CARLA we can not change that.
Then, because it uses only on thread for that, is normal than one of your cores will get more busy than others, because the process can not be distributed like others.

**Reply by ebinroyabraham** [reply]
Hi,
Many thanks for your reply.
